cpp/src/gis/gdal/gis_functions.cpp

```diff
@@ -220,6 +242,24 @@ inline std::shared_ptr<arrow::Array> AppendWkb(
   return array_ptr;
 }
 
+inline void AppendWkb(arrow::BinaryBuilder& builder, const OGRGeometry* geo) {
+  if (geo == nullptr) {
+    CHECK_ARROW(builder.AppendNull());
+  } else if (geo->IsEmpty() && (geo->getGeometryType() == wkbPoint)) {
+    CHECK_ARROW(builder.AppendNull());
+  } else {
+    auto wkb_size = geo->WkbSize();
+    auto wkb = static_cast<unsigned char*>(CPLMalloc(wkb_size));
+    auto err_code = geo->exportToWkb(OGRwkbByteOrder::wkbNDR, wkb);
+    if (err_code != OGRERR_NONE) {
+      CHECK_ARROW(builder.AppendNull());
+    } else {
+      CHECK_ARROW(builder.Append(wkb, wkb_size));
+    }
+    CPLFree(wkb);
+  }
+}
+
```

Comment:
what's the difference between `AppendWkb` and `AppendWkbNDR`,  It looks they have the same code ?

Reply:
You're right. This AppendWkb will be deleted.